### DISCUSSION TITLE
Time complexity of Jacobian computation

### DISCUSSION THREAD

**Discussion by bharswami** [autodiff/autodiff#341] [General]
What is the time-complexity of Jacobian computation for mx1 vector and nx1 variables. Is it O(m*n)?

**Comment by allanleal** [comment]
Using dual or real, for _m_ functions in _n_ variables, the _m_ functions need to be evaluated _n_ times, once for each seeded variable. Thus, theoretically, O(m*n). In practice, however, there are problems in which not all functions depend on all variables. In such cases, it's possible to avoid function re-evaluations. This optimization is problem-dependent and has to be done manually by the user.
